lib/fluent/plugin/out_mysql_bulk.rb

```diff
@@ -145,7 +145,7 @@ def write(chunk)
         data = format_proc.call(tag, time, data, max_lengths)
         values << Mysql2::Client.pseudo_bind(values_template, data)
       end
-      sql = "INSERT INTO #{table} (#{@column_names.join(',')}) VALUES #{values.join(',')}"
+      sql = "INSERT INTO `#{table}` (#{@column_names.map { |colname| "`#{colname}`" }.join(',')}) VALUES #{values.join(',')}"
```

Comment:
Column names [may](https://dev.mysql.com/doc/refman/5.7/en/identifiers.html) legally include a backtick character themselves.

> Identifier quote characters can be included within an identifier if you quote the identifier. If the character to be included within the identifier is the same as that used to quote the identifier itself, then you need to double the character. The following statement creates a table named a`b that contains a column named c"d:
```sql
CREATE TABLE `a``b` (`c"d` INT);
```
So in addition to wrapping names in backticks, you will also have to replace backticks inside the names with double backticks.